process/peer/process.go

```diff
@@ -338,24 +339,12 @@ func (vs *validatorStatistics) checkForMissedBlocks(
 	}
 
 	tooManyComputations := missedRounds > vs.maxComputableRounds
-	if tooManyComputations {
-		consensusGroup, err := vs.nodesCoordinator.ComputeValidatorsGroup(prevRandSeed, previousHeaderRound+1, shardId)
-		if err != nil {
-			return err
-		}
-
-		consensusGroupSize := len(consensusGroup)
-		err = vs.decreaseAll(shardId, missedRounds, consensusGroupSize)
-		if err != nil {
-			return err
-		}
-	} else {
-		err := vs.computeDecrease(previousHeaderRound, currentHeaderRound, prevRandSeed, shardId)
-		if err != nil {
-			return err
-		}
+	if !tooManyComputations {
+		return vs.computeDecrease(previousHeaderRound, currentHeaderRound, prevRandSeed, shardId)
 	}
-	return nil
+
+	consensusGroupSize := vs.nodesCoordinator.ConsensusGroupSize(shardId)
```

Comment:
Now, I would move this line inside decreaseAll method and I would remove its third parameter

Reply:
yes! fixed